spec/features/admin/show_merchants_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe "admin" do
+  describe "can see all merchants through the all merchants page" do
+    it "and can disable a merchant as well as their items" do
+      admin = User.create!(name: "bob", password: '12345', password_confirmation: '12345', address: "street", city: "Denver", state: "CO", zip:"12345", email: "someone@gmail.com", role: 2)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(admin)
+
+      5.times do
+        shop = create(:merchant)
+        item = create(:item, merchant_id: shop.id)
+        item = create(:item, merchant_id: shop.id)
+      end
+      merchant1 = Merchant.all.first
+      merchant2 = Merchant.all.last
+
+      visit '/'
+
+      click_link 'All Merchants'
+
+      expect(page).to have_content(merchant1.name)
+      expect(page).to have_content(merchant2.name)
+
+      within "#merchant#{merchant1.id}" do
+        click_link "Disable Merchant"
+      end
+
+      expect(page).to have_content("#{merchant1.name} has been disabled.")
+      expect(current_path).to eq('/admin/merchants')
+
+      item_statuses = merchant1.items.pluck(:active?).uniq
```

Comment:
Really cool way of testing this with AR as opposed to checking the page content.